### DISCUSSION TITLE
Combobox with grouping support

### DISCUSSION THREAD

**Discussion by kumar-tadepalli** [vmware-clarity/ng-clarity#635] [Q&A] [answered]
Hi,

Is it possible to provide grouping in combobox component?

Something like 
![image](https://user-images.githubusercontent.com/12848838/231207953-bd844521-fd38-4c33-87f8-3848f75ada37.png)

**Comment by kumar-tadepalli** [comment]
Can some one from clarity team confirm on this?

**Reply by kevinbuhmann** [reply]
Yes, this is possible since you can project arbitrary content into the `clr-options` component. See this example: https://stackblitz.com/edit/clarity-light-theme-clr15-pweiue?file=src%2Fapp%2Fapp.component.html

**Reply by kumar-tadepalli** [reply]
Thanks for the response.

I see two issues here.
1. The search is not working in the given stack blitz example.
2. You need a wrapper DOM element to apply *ngFor like opt-group
For select  component we can write this way:
```
<select id="select-full"
        class="clr-select"
        formControlName="vms"
        required>
	<optgroup *ngFor="let vm of vms| keyvalue"
	          [label]="getCategoryNameForOptGroup(vm.key)">
		<option *ngFor="let disk of vm.value"
		        [ngValue]="disk">
                    {{ disk.name }}
                  </option>
	</optgroup>
</select>
``` 
Can help me with nested iteration syntax ?

**Reply by kevinbuhmann** [reply]
1. The filter/search on input requires using `*clrOptionItems`. I don't think there is a way to make that work with grouping.
2. You can use `ng-container` for the group iterator.

**Reply by kevinbuhmann** [reply]
Update on filtering: I found a solution. It's kinda hacky (uses a private member of the `ClrOptionItems` directive class), but it works.

https://stackblitz.com/edit/clarity-light-theme-clr15-pai9k5?file=src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html

**Reply by kevinbuhmann** [reply]
This still needs work, though. It's no longer keyboard-accessible (options cannot be selected via the up/down arrow keys). I think that will require a library update.

**Reply by kumar-tadepalli** [reply]
Keyboard accessibility should be needed for end users. I will create a feature request as this is basic feature in today's web.
Thanks for the help kevin. 👍

**Reply by kevinbuhmann** [reply, marked as answer]
I have a proposal already in #639. I will be discussing it with my team.

**Reply by kumar-tadepalli** [reply]
@kevinbuhmann Any expected date for this feature?

**Reply by kevinbuhmann** [reply]
No, not at this time.
